tests/ci/cdk/pipeline/ci_stage.py

```diff
@@ -0,0 +1,287 @@
+                    environment_variables={
+                        "GITHUB_PAT":  codebuild.BuildEnvironmentVariable(
+                            type=codebuild.BuildEnvironmentVariableType.SECRETS_MANAGER,
+                            value=GITHUB_TOKEN_SECRET_NAME,
+                        ),
+                    }
+                ),
+                commands=[
+                    "env",
+                    "curl -H \"Authorization: token ${GITHUB_PAT}\" https://api.github.com/user",
+                    "git clone https://${GITHUB_PAT}@github.com/${STAGING_GITHUB_REPO_OWNER}/${STAGING_GITHUB_REPO_NAME}.git",
+                    "git remote add upstream https://github.com/aws/aws-lc.git",
+                    "git fetch upstream",
+                    "git checkout main",
+                    "git merge upstream/main",
+                    # "git push origin main",
+                ],
+                env={
+                    "STAGING_GITHUB_REPO_OWNER": STAGING_GITHUB_REPO_OWNER,
+                    "STAGING_GITHUB_REPO_NAME": STAGING_GITHUB_REPO_NAME,
+                },
+                role=role,
+                timeout=Duration.minutes(60),
+            )
+
+        env = env or {}
+
+        prebuild_check_step = pipelines.CodeBuildStep(
+            "PrebuildCheck",
+            input=input,
+            commands=[
+                "cd tests/ci/cdk/pipeline/scripts",
+                "chmod +x check_trigger_conditions.sh",
+                'trigger_conditions=$(./check_trigger_conditions.sh --build-type ci --stacks "${STACKS}")',
+                "export NEED_REBUILD=$(echo $trigger_conditions | sed -n 's/.*\(NEED_REBUILD=[0-9]*\).*/\\1/p' | cut -d'=' -f2 )",
+            ],
+            env={
+                **env,
+                "STACKS": " ".join(stack_names),
+            },
+            role=role,
+            timeout=Duration.minutes(60),
+        )
+
+        batch_build_jobs = {
+            "build-list": [
+                {
+                    "identifier": options.identifier,
+                    "ignore-failure": options.ignore_failure,
+                    "env": {
+                        "variables": {
+                            "PROJECT": options.project,
+                            "TIMEOUT": str(max_retry * options.timeout),
+                            **options.env,
+                        }
+                    },
+                }
+                for options in self.build_options
+            ]
+        }
+
+        ci_run_step = CodeBuildBatchStep(
+            f"BuildStep",
+            action_name="StartWait",
+            input=input,
+            commands=[
+                "cd tests/ci/cdk/pipeline/scripts",
+                "chmod +x build_target.sh",
+                "./build_target.sh --build-type ci --project ${PROJECT} --max-retry ${MAX_RETRY} --timeout ${TIMEOUT}",
+            ],
+            role=role,
+            timeout=300,
```

Comment:
What's the relation to this timeout value here, and the one computed on line 236, and then passed to the script on line 252?